### PR TITLE
Mt

### PR DESCRIPTION
**Pull request by marceltreptow**
Add 
- Malte Seeker
- Enemy count
- Level countdown

### CODE REVIEW

File: src/engine/core/gamemanagement/GameSystems.java [hunk 1]
```diff
@@ -1463,41 +1557,62 @@ private void shoot(EntityManager manager, Vector3 direction, int id, float speed
         }
 
         private void calculateCollision(EntityManager manager, int id, float deltatime) {
```
Comment (kastanileel) [inline src/engine/core/gamemanagement/GameSystems.java:1559]: why remove the null check? this prevents errors?

File: src/engine/core/gamemanagement/GameSystems.java [hunk 2]
```diff
@@ -696,7 +701,7 @@ private void shoot(EntityManager manager, int id, Vector3 direction, float speed
                     manager.rendering[bulletId].renderType = GameComponents.Rendering.RenderType.OneColor;
                     manager.rendering[bulletId].modelRotation = new Vector3(0.0f, 3.1415f / -2.0f, 0.0f);
                     manager.physicsBody[bulletId].speed = speed;
-                    manager.bullet[bulletId].lifeTime = bulletLifeTime;
+                    manager.bullet[bulletId].lifeTime = lifeTime;
```
Comment (kastanileel) [inline src/engine/core/gamemanagement/GameSystems.java:704]: should be bulletLifeTime

File: src/engine/core/gamemanagement/GameSystems.java [hunk 3]
```diff
@@ -420,45 +422,48 @@ private void doShooting(EntityManager manager, int id, float deltaTime) {
...
                             magazineShotgun = 2;
                         }
                     }
 
                 }
-                case SNIPER ->{
+                case SNIPER -> {
                     DrawingWindow.currentAmmo = magazineSniper;
-                    if(MMouseListener.getInstance().isLeftButtonPressed()  || MKeyListener.getInstance().isKeyPressed('u') || MKeyListener.getInstance().isKeyPressed('u')){
+                    if (MMouseListener.getInstance().isLeftButtonPressed() || MKeyListener.getInstance().isKeyPressed('u') || MKeyListener.getInstance().isKeyPressed('u')) {
```
Comment (kastanileel) [inline src/engine/core/gamemanagement/GameSystems.java:458]: should be one upper case U

**Review by kastanileel — CHANGES_REQUESTED** [review]

File: src/engine/core/gamemanagement/GameSystems.java [hunk 4]
```diff
@@ -647,14 +652,14 @@ private void handleKnife(EntityManager manager, int id){
...
-        private void reactToKnifeCollision(EntityManager manager, int knife, int otherId){
+        private void reactToKnifeCollision(EntityManager manager, int knife, int otherId) {
             GameComponents.Collider.ColliderTag tag = manager.collider[otherId].colliderTag;
             switch (tag) {
                 case ENEMY -> {
                     manager.damageable[otherId].health -= 50;
                     System.out.println("Enemy health: " + manager.damageable[otherId].health);
-                    if(!DamageSystem.damagedEntities.contains(otherId))
-                        DamageSystem.damagedEntities.add(otherId);
+
```
[the remark this answers is not in the document]
Comment (marceltreptow) [reply src/engine/core/gamemanagement/GameSystems.java:661]: Probably was deleted while merging with main

File: src/engine/core/gamemanagement/GameSystems.java [hunk 5]
```diff
@@ -737,16 +742,19 @@ private void reactToCollisionTag(EntityManager entityManager, int playerId, int
                         } catch (Exception e) {
                             System.out.println("oops");
                         }
+                        if (entityManager.pickupWeapon[otherId].weaponType != GameComponents.PlayerMovement.WeaponType.HEALTHPACK)
+                            MusicPlayer.getInstance().playSound(MusicPlayer.SoundEffect.BIGGER_GUN);
```
[the remark this answers is not in the document]
Comment (marceltreptow) [reply src/engine/core/gamemanagement/GameSystems.java:746]: I forget to change it back

File: src/engine/core/gamemanagement/GameSystems.java (hunk 1 above)
Comment (marceltreptow) [reply src/engine/core/gamemanagement/GameSystems.java:1559]: For debugging. I change it back

File: src/engine/core/gamemanagement/GameSystems.java (hunk 3 above)
Comment (marceltreptow) [reply src/engine/core/gamemanagement/GameSystems.java:458]: Good that you noticed because it is the same in main

**Review by kastanileel — APPROVED** [review]
👍

**Pull request merged**